### PR TITLE
prov/sockets: Update NEWS.md for 1.4 release

### PR DESCRIPTION
**Pull request by shantonu**
@jithinjosepkl 

Signed-off-by: Shantonu Hossain <shantonu.hossain@intel.com>

### CODE REVIEW

File: NEWS.md [hunk 1]
```diff
@@ -24,6 +24,39 @@ v1.4.0, TBD
...
 
+## Sockets provider notes
+
+- General code cleanup and bug fixes
+- Set tx/rx op_flags correctly to be consistent with manpage
+- Restructure struct sock_ep to support alias ep
+- Refactor CQ/Cntr bindings, CQ completion generation, and counter increments
+- Copy compare data to internal buffer when FI_INJECT is set in fi_compare_atomic
+- Correctly handle triggered operation when FI_INJECT is set or triggered op is
+enqueued or counter is incremented. Initilize counter threshold to INT_MAX
```
Comment (jithinjosepkl) [inline NEWS.md]: initialize

File: NEWS.md [hunk 2]
```diff
@@ -24,6 +24,39 @@ v1.4.0, TBD
...
+only acquired once per iteration
+- Refactor connection management of MSG ep so that it uses TCP instead of UDP for
+connection management msg and new port for every MSG endpoint
+- Add sock_cq_sanitize_flags function to make sure only flags returned in CQ events
+are the ones that are listed on the manpage
+- Update fi_poll semantics for counters so that it returns success if the counter
+value is different from the last-read-value
+- Allow multiple threads to wait on one counter
+- Update code to use ofi_util_mr - the new MR structure added to util code
+- Fix fi_av_insert not to report error when # of inserted addr exceeds the count
```
Comment (jithinjosepkl) [inline NEWS.md]: # -> number?

**Comment by jithinjosepkl** [comment]
👍

**Comment by shefty** [comment]
Replace by PR #2400 - to fix merge conflict and add slight formatting changes.

**Pull request closed without merging**